Add/Restore the option to automatically boot into Debug Warp Screen

nclok1405 opened the pull request:
This Pull Request adds (back) the ability to directly launch into the Debug Warp Screen when launching or resetting the game.

This is intended only for developer/quick testing use, so the toggle is located at Dev Tools section and require the Debug Mode to have any effect.

<!--- section:artifacts:start -->
### Build Artifacts
  - [soh-linux.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/3321014984.zip)
  - [soh-windows.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/3321030756.zip)
  - [soh-mac.zip](https://nightly.link/HarbourMasters/Shipwright/actions/artifacts/3321046877.zip)
<!--- section:artifacts:end -->

nclok1405 committed:
Added/Restored the option to automatically boot into Debug Warp Screen

nclok1405 committed:
clang-formated

nclok1405 commented:
I forgot to run clang-format... I request reopening the PR.

Pepper0ni commented:
You currently get forced into the debug warp screen when you boot file 1 in debug mode, does this override that functionality (which i assume is vanilla?)

nclok1405 commented:
> You currently get forced into the debug warp screen when you boot file 1 in debug mode, does this override that functionality (which i assume is vanilla?)

"debug warp screen when you boot file 1 in debug mode" is indeed a [vanilla Debug ROM behavior](https://tcrf.net/Proto:The_Legend_of_Zelda:_Ocarina_of_Time_Master_Quest/Map_Select#Debug_ROM). This Pull Request does not change that behavior.

Pepper0ni commented on soh/src/overlays/gamestates/ovl_file_choose/z_file_choose.c:
```diff
@@ -3586,6 +3586,14 @@ void FileChoose_Main(GameState* thisx) {
     }
 
+    // Boot to Debug Warp Screen
+    if ((CVarGetInteger(CVAR_DEVELOPER_TOOLS("DebugEnabled"), 0) != 0) &&
```
This should be done in a hook

Pepper0ni reviewed (dismissed):
It feels a little weird to have this setting separate from the boot screen controls, but it makes some sense as a debug setting, so it should be fine.

nclok1405 committed:
Added a new hook and moved Boot To Debug Warp Screen to it

nclok1405 committed:
clang

nclok1405 committed:
Added DebugEnabled to initFunc's CVar list. This should prevent Debug Warp from being triggered when Boot to Debug Warp option is enabled but Debug Mode option is disabled.

nclok1405 committed:
No longer hijacks CustomLogoTitle

nclok1405 commented:
I've added a new hook and moved most of Boot to Debug Warp Screen code to the newly added hook.

As for toggle location, I think it should remain separated from Boot Sequence dropdown list (then known as "Fast File Select") because it was "overcomplicated" (#4882). The ability to boot into the Debug Warp Screen was removed during Boot Sequence option redesign, but it is very useful for quick testing and development usage, so this ability should return, just hidden behind Dev Tools tab.

nclok1405 committed:
Disable "Boot Sequence" dropdown when Boot to Debug Warp Screen is enabled

Malkierian reviewed (approved):
Love to see someone using the disabledMap.
